plugins/deisa/deisa.cxx

```diff
@@ -0,0 +1,228 @@
+				py::finalize_interpreter();
+			}
+		} catch (const std::exception& e) {
+			context().logger().error("Exception in destructor, caught exception {}", e.what());
+		} catch (...) {
+			context().logger().error("Exception in destructor");
+		}
+	}
+
+private:
+	/**
+	 * Check that the PDI plugin is compatible with Deisa's python Bridge (i.e, that the python API is what it should be).
+	 * The plugin is compatible if PYTHON_LIBRARY_COMPATIBILITY == deisa.__version__.__version__ (python)
+	 */
+	static void check_compatibility()
+	{
+		pymod deisa = pymod::import("deisa.__version__");
+		const auto python_library_version = py::str(deisa.attr("__version__")).cast<std::string>();
+		if (python_library_version != PYTHON_LIBRARY_COMPATIBILITY) {
+			throw Plugin_error(
+				"Deisa PDI plugin is expecting Deisa python version " + std::string(PYTHON_LIBRARY_COMPATIBILITY) + " but found version "
+				+ python_library_version
+			);
+		}
+	}
+
+	void init_deisa()
+	{
+		std::unordered_map<std::string, std::unordered_map<std::string, std::vector<size_t>>> darrs;
+		std::unordered_map<std::string, py::dtype> darrs_dtype;
+		for (auto&& key_value: m_deisa_arrays) {
+			std::unordered_map<std::string, std::vector<size_t>> darr;
+			std::vector<size_t> sizes;
+			std::vector<size_t> starts;
+			std::vector<size_t> subsizes;
+			std::vector<size_t> timedim;
+			std::string deisa_array_name = key_value.first;
+			Datatype_sptr type_sptr = key_value.second->evaluate(context());
+			timedim.emplace_back(key_value.second->attribute("timedim").to_long(context()));
+			// get info from datatype
+			while (auto&& array_type = std::dynamic_pointer_cast<const PDI::Array_datatype>(type_sptr)) {
+				sizes.emplace_back(array_type->size());
+				starts.emplace_back(array_type->start());
+				subsizes.emplace_back(array_type->subsize());
+				type_sptr = array_type->subtype();
+			}
+			darr["sizes"] = sizes;
+			darr["starts"] = starts;
+			darr["subsizes"] = subsizes;
+			darr["timedim"] = timedim;
+			darrs[deisa_array_name] = darr;
+			darrs_dtype[deisa_array_name] = datatype_to_pydtype(std::dynamic_pointer_cast<const Scalar_datatype>(type_sptr));
+		}
+
+		try {
+			// setup python context and instantiate bridge
+			pydict pyscope = pymod::import("__main__").attr("__dict__");
+			pyscope["deisa"] = pymod::import("deisa");
+			pymod deisa = pymod::import("deisa");
+			pyscope["get_bridge_instance"] = deisa.attr("get_bridge_instance");
+			pyscope["scheduler_info"] = to_python(m_scheduler_info.to_ref(context()));
+			pyscope["rank"] = to_python(m_rank.to_ref(context()));
+			pyscope["size"] = to_python(m_size.to_ref(context()));
+			pyscope["deisa_arrays"] = darrs;
+			pyscope["deisa_arrays_dtype"] = darrs_dtype;
+
+			// TODO: use_ucx
+			py::exec("bridge = get_bridge_instance(scheduler_info, rank, size, deisa_arrays, deisa_arrays_dtype)", pyscope);
```

Comment:
I'd use class variables rather than python global variables so as not to pollute the global namespace.
```suggestion
			m_deisa = pymod::import("deisa");
			py::object get_bridge_instance = m_deisa.attr("get_bridge_instance");
			
			// TODO: use_ucx
			py::object bridge = get_bridge_instance(
				to_python(m_scheduler_info.to_ref(context())),
				to_python(m_rank.to_ref(context())),
				to_python(m_size.to_ref(context())),
				darrs,
				darrs_dtype);
```